### PR TITLE
fix: add logging to alerts and reports to find non-triggering issues

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
Alerts and reports has had some issues where some users report their reports not triggering at their validate time. This pr is the first step in adding some logs to gather error feedback to correct this problem.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/reports/commands/alert.py [hunk 1]
```diff
@@ -166,6 +166,9 @@ def _execute_query(self) -> pd.DataFrame:
             logger.warning("A timeout occurred while executing the alert query: %s", ex)
             raise AlertQueryTimeout() from ex
         except Exception as ex:
+            logger.warning(
+                "A general exception occurred while executing the alert query: %s", ex
+            )
             raise AlertQueryError(message=str(ex)) from ex
```
Comment (eschutho) [inline superset/reports/commands/alert.py:172]: does this exception already log this error?

File: superset/reports/commands/execute.py [hunk 2]
```diff
@@ -436,6 +436,7 @@ def send_error(self, name: str, message: str) -> None:
         """
         header_data = self._get_log_data()
         header_data["error_text"] = message
+        logger.info("header_data infor %s", header_data)
```
Comment (eschutho) [inline superset/reports/commands/execute.py:439]: typo `infor`

File: superset/reports/commands/alert.py (hunk 1 above)
Comment (pkdotson) [reply superset/reports/commands/alert.py:172]: Yes but i'm curious to see what this error is on it's face because it could be empty.

**Review by eschutho — APPROVED** [review]

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.39

**Pull request merged**